### DISCUSSION TITLE
Feature Request regarding Color Management (embedded Color Profiles)

### DISCUSSION THREAD

**Discussion by blackfoxx79** [d2phap/ImageGlass#1593] [General]
Hi there.
I'm a Photographer for 30 Years now and was always looking for an simple Picture Viewer, which shows Informations about embedded Color Profiles at first Sight. So I was very glad to find ImageGlass, cause it seems to be the only One which does this.
**THANK YOU VERY MUCH** for that!
Before ImageGlass I had to open the Pics one after another with GIMP, press "B" and click on Tab "Color Profile" to get the Information I needed. Which **was** a very annoying and time consuming "Process"...
Background:
As I upgraded my RawTherapee Installation from 5.8 to 5.9 some months ago, there was a Misconfiguation which led to NOT embed the chosen Output Color Profile to the exported JPGs. But during my usual Workflow I never recognized it. I just noticed that there was something wrong with the Colors when I had the same picture opened in RawTherapee and IrfanView (my former Picture Viewer) at the same Time. The Differences were minimal, but enough to start some Research. Which took me some time to find out the Reason why (no embedded Color Profile).
Fortunately only about hundred Photos were affected. So it wasn't a high Effort to export them again WITH the correct embedded Color Profiles.
But then I started looking for an Picture Viewer that shows Color Profile Informations at first sight, to prevent such mistakes...
Now my Request:
I noticed that ImageGlass shows only some kind of "reduced" Info about embedded Color Profiles in the Top Bar.
For Example "sRGB" or no Field if there's no Color Profile embedded.
My Wish is to have some more "extended" Information there.
For Example "No Color Profile" (as Text), if there's no Color Profile embedded. Or the complete Name of the embedded Color Profile, like GIMP does. (See attached Screenshots.)
Because I'm working with half dozen Color Profiles (for Display only Presentation, for WWW Uploads, for Printing with different Printers, etc...). So it would be really great to get all the Infos I need while just scrolling through Picture Folders with less Extra-Clicks as possible.
Kind Regards :)

embedded Color Profile Informations within GIMP:
![GIMP_RTv4-sRGB](https://github.com/d2phap/ImageGlass/assets/61013245/72587e73-5a2b-4cfd-9fb6-80414e28385d)
![GIMP_Adobe-RGB](https://github.com/d2phap/ImageGlass/assets/61013245/aaa5deb0-1afd-48ea-91d6-50f5b4473451)

"RTv4_sRGB" is only shown as "sRGB" in ImageGlass:
![ImageGlass_sRGB_Color-Profile](https://github.com/d2phap/ImageGlass/assets/61013245/4b1f133b-da57-481d-aa33-739cf1d30a85)

No Color Profile Field if Pic has no Color Profile in ImageGlass:
![ImageGlass_No_Color-Profile](https://github.com/d2phap/ImageGlass/assets/61013245/2a59c0d7-95e9-42d6-914d-64a0a8c8e9fc)
Some more extended Information about embedded or no Color Profile would be great.

**Comment by d2phap** [comment]
To view image metadata you can use Exif tool (main menu > Tools > Exif tool, or press X)

![image](https://github.com/d2phap/ImageGlass/assets/3154213/4d0cae9f-19ef-425f-b66f-de4240584871)

You can add these settings to the Exif tool (in Settings > Tools tab) to show only the color profile tags like GIMP:
```
-ProfileDescription -ColorSpaceData -DeviceManufacturer -DeviceModel
```

![image](https://github.com/d2phap/ImageGlass/assets/3154213/d01c388f-cfa4-4aa1-8003-326cc03d9d03)

**Reply by blackfoxx79** [reply]
Thanks for this hint!
I tried and it works as expected.
The only downside with this method is that it needs another window to be open.
But for now it should be sufficient enough for me.
Although it would be very nice to have this feature (like I wished) implemented somewhen in the future.
Regards

**Comment by koscelansky** [comment]
Hi, I started using ImageGlass couple of days ago and I immediately though this must be a bug. Then I quick search issues and found nothing. So I almost created new one, but then I check discussions. This bugs me out, what is the usefulness of it saying sRGB? It just annoy me, because I accidentally freak out that I screw something in image processing and my Adobe RGB files are now sRGB. Only to realize that it is correct and IG just show something not very accurate in my opinion. I even debug the code little bit and found out where the color space data is generated (see attached screenshot). As you can see the info I want is just there. Could it be maybe possible to change the data shown in the title to be the Description instead of ColorSpace? If not maybe add an option to hide that info from title I found it misleading, however I will probably adjust to ignore it, still it would be nice to have there the info I want :). 

 
![Screenshot 2024-05-09 010535](https://github.com/d2phap/ImageGlass/assets/24921049/ea435687-0608-4b9c-a64c-3b579860eb37)

**Reply by d2phap** [reply]
OK the next version will show `Description`

**Reply by d2phap** [reply]
@koscelansky [ImageGlass 9.1 Beta](https://github.com/d2phap/ImageGlass/releases/tag/9.0.11.529) now showing the Color Profile "Description" on the title bar and the thumbnail tooltip

![image](https://github.com/d2phap/ImageGlass/assets/3154213/74e07e08-c1a7-4419-a904-063e369786d3)

![image](https://github.com/d2phap/ImageGlass/assets/3154213/ebf8a56d-fc6a-4fb2-8aa8-5c72c0daf3e8)

**Reply by PaulCoddington** [reply]
In the screen capture above, ImageGlass is displaying both sRGB and U.S. Web Coated SWOP at the same time. The image format can only be RGB or CMYK, it cannot be both.

Unless sRGB is the color space of the monitor? As in, the intention is "this is U.S. Web Coated SWOP CMYK translated by color management to sRGB because the current monitor profile is sRGB".

Even so, on my system with 9.1.8.x, all I see is "sRGB" regardless of monitor or image profile.

Also, the smart date feature seems to return the file modified date. So, it's not clear how to display the date a photo was taken.

The example above is JPEG. Could it be that ImageGlass is not retrieving embedded metadata correctly for other formats, such as PNG?

Perhaps running some tests with, say, a Display P3 ICC profiled PNG with embedded XMP metadata might shed light on the problem?

**Reply by d2phap** [reply]
These values are retrieved from Magick.NET, see [the screenshot](https://github.com/d2phap/ImageGlass/discussions/1593#discussioncomment-9361434). There are 3 pieces of information displayed:
- The red rectangle: is the color space of the image (see [`ImageMagick.ColorSpace`](https://github.com/d2phap/ImageGlass/blob/eda465af4f340eb93ec957cb62a1b7e62e82c7d5/Source/Components/ImageGlass.Base/Photoing/Codecs/PhotoCodec.cs#L122))
- The green rectangle: is the color profile name of the image (see [`ImageMagick.ColorProfile.Description`](https://github.com/d2phap/ImageGlass/blob/eda465af4f340eb93ec957cb62a1b7e62e82c7d5/Source/Components/ImageGlass.Base/Photoing/Codecs/PhotoCodec.cs#L205-L213))
- The yellow rectangle: is the color space of the color profile (see [`ImageMagick.ColorProfile.ColorSpace`](https://github.com/d2phap/ImageGlass/blob/eda465af4f340eb93ec957cb62a1b7e62e82c7d5/Source/Components/ImageGlass.Base/Photoing/Codecs/PhotoCodec.cs#L205-L213))
If the image does not contain a color profile (`ImageMagick.ColorProfile`), only the red rectangle displayed

Magick.NET does not fully support image metadata. You can use [ExifGlass tool](https://github.com/d2phap/ExifGlass) to verify the color information of your image. Without provided image file, I cannot check further.

![image](https://github.com/user-attachments/assets/ed108082-3862-4ccd-8c12-1e807d5d1830)

**Reply by blackfoxx79** [reply]
Thanks for your replies and effort, but meanwhile I switched to "XNview" as my default photo viewer...
It does exactly what I want and need, it's slim, doesn't need "extra software" or dependencies to be installed beside and it's completely customizable to fulfill any photographers needs...
Anyway, I wish you good luck with ImageGlass furthermore!
(I don't close this discussion for now, as it might be helpful for someone else in future.)
Kind regards.
